test/integration/controller.v2/trainjob_controller_test.go

```diff
@@ -71,4 +73,120 @@ var _ = ginkgo.Describe("TrainJob controller", ginkgo.Ordered, func() {
 			gomega.Expect(k8sClient.Create(ctx, trainJob)).Should(gomega.Succeed())
 		})
 	})
+
+	ginkgo.When("TrainJob CR Validation", func() {
```

Comment:
Ideally, we should more strictly error check by custom gomega Error matcher.
But, after the baseline implementation, we can improve this testing.

So, for now, I'm ok with this non-strictly checks.

Reply:
Created https://github.com/kubeflow/training-operator/issues/2293